src/rootPages/Designer/properties/views/ABViewDocxBuilder.js

```diff
@@ -0,0 +1,405 @@
+                              this.onChange();
+                           },
+                        },
+                     },
+                  ],
+               },
+            },
+         ]);
+      }
+
+      populate(view) {
+         super.populate(view);
+         if (!view) return;
+
+         const ids = this.ids;
+
+         const $DcSelector = $$(ids.datacollection);
+
+         const selectedDvId = view.settings.dataviewID ?? null;
+
+         $$(ids.toolbarBackground).setValue(
+            view.settings.toolbarBackground ??
+               ABViewDocxBuilderPropertyComponentDefaults.toolbarBackground
+         );
+         $$(ids.buttonPosition).setValue(
+            view.settings.buttonPosition ??
+               ABViewDocxBuilderPropertyComponentDefaults.buttonPosition
+         );
+
+         // Pull data views to options
+         const dcOptions = view.application
+            .datacollectionsIncluded()
+            .map((dc) => {
+               return {
+                  id: dc.id,
+                  value: dc.label,
+               };
+            });
+
+         $DcSelector.define("options", dcOptions);
+         $DcSelector.define("value", selectedDvId);
+         $DcSelector.refresh();
+
+         $$(ids.language).setValue(
+            view.settings.language ??
+               ABViewDocxBuilderPropertyComponentDefaults.language
+         );
+
+         $$(ids.filelabel).setValue(view.filelabel ?? view.settings.filelabel);
+         $$(ids.buttonlabel).setValue(
+            view.buttonlabel ?? view.settings.buttonlabel
+         );
+         $$(ids.width).setValue(view.settings.width);
+
+         if (view.settings.filename) {
+            $$(ids.docxDownload).show();
+         } else {
+            $$(ids.docxDownload).hide();
+         }
+      }
+
+      defaultValues() {
+         let values = {};
+         const ViewClass = this.ViewClass();
+         if (ViewClass) {
+            values = ViewClass.defaultValues();
+         }
+         return values;
+      }
+
+      /**
+       * @method values
+       * return the values for this form.
+       * @return {obj}
+       */
+      values() {
+         const ids = this.ids;
+         let vals = super.values();
+
+         vals.settings = vals.settings ?? {};
+         vals.settings.buttonlabel = $$(ids.buttonlabel).getValue();
+         vals.settings.dataviewID = $$(ids.datacollection).getValue();
+         vals.settings.width = $$(ids.width).getValue();
+         vals.filelabel = $$(ids.filelabel).getValue();
+         vals.settings.language = $$(ids.language).getValue();
+         vals.settings.toolbarBackground = $$(ids.toolbarBackground).getValue();
+         vals.settings.buttonPosition = $$(ids.buttonPosition).getValue();
+
+         return vals;
+      }
+
+      /**
+       * @method FieldClass()
+       * A method to return the proper ABViewXXX Definition.
+       * NOTE: Must be overwritten by the Child Class
+       */
+      ViewClass() {
```

Comment:
Is it `FieldClass` or `ViewClass`? Comment doesn't match.

Reply:
Hawk eyes.

Thank you.